lib/galaxy/model/__init__.py

```diff
@@ -846,7 +846,7 @@ def get_param_values( self, app ):
         dict of tool parameter values.
         """
         param_dict = dict( [ ( p.name, p.value ) for p in self.parent_job.parameters ] )
```

Comment:
Not connected with your changes, but I think `self.parent_job` should be `self.job` .
Introduced in commit 988d94f2d43812ee590dfd169fb0d4ab33f31036, ping @dannon.